dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -213,150 +208,116 @@ StaticLinkageCheckReport findLinkageErrors() throws ClassNotFoundException, IOEx
     return StaticLinkageCheckReport.create(jarLinkageReports.build());
   }
 
-  private JarLinkageReport findInvalidReferences(
-      Path jarPath, SymbolReferenceSet symbolReferenceSet) {
+  /**
+   * Generates a linkage report for a jar file, by checking linkage errors in the symbol
+   * references against the input class path.
+   *
+   * @param jarPath absolute path to the jar file
+   * @param symbolReferenceSet symbol references from {@code jarPath} to check its linkage errors
+   * @return linkage report for the jar file, which includes linkage errors if any
+   */
+  @VisibleForTesting
+  JarLinkageReport generateLinkageReport(Path jarPath, SymbolReferenceSet symbolReferenceSet) {
     JarLinkageReport.Builder reportBuilder = JarLinkageReport.builder().setJarPath(jarPath);
 
-    // TODO(suztomo): implement validation for field, method and class references in the table
-    throw new UnsupportedOperationException("The report generation is not yet implemented");
+    // As the Java compiler ensures that there is no static linkage errors between classes defined
+    // in the same jar file, this validation excludes reference within the same jar file.
```

Comment:
file, this --> file. This

Reply:
The first "As" means "Because". Maybe I'll change the sentence to

> Because the java compiler...in the same jar file, this validation